### PR TITLE
Implement image upload

### PR DESCRIPTION
**Pull request by rshipp**
Task: https://app.asana.com/0/1170776727341290/1172626862010553/f

I'm not sure how to rspec the controller for this, @sadleb any suggestions?

### CODE REVIEW

**Comment by sadleb** [comment]
> I’ve been punting on controller specs too. I need to spend more time researching. So, no ideas, but I would queue up a task to write a controller spec for this. That’s what I’ve been doing.
> 
> That said, this is way cleaner than I expected!

Oh wait, I just noticed the hard coded course id = 1. Let’s introduce a constant for content library.

**Comment by sadleb** [comment]
> > I’ve been punting on controller specs too. I need to spend more time researching. So, no ideas, but I would queue up a task to write a controller spec for this. That’s what I’ve been doing.
> > That said, this is way cleaner than I expected!
> 
> Oh wait, I just noticed the hard coded course id = 1. Let’s introduce a constant for content library.

Eg CONTENT_LIBRARY_COURSE_ID

**Comment by rshipp** [comment]
@sadleb I used `ContentLibraryCourseID` because it matches the other constants in that file and I don't know ruby formatting rules. lmk if it should be all-caps and/or somewhere else.

**Comment by sadleb** [comment]
> @sadleb I used `ContentLibraryCourseID` because it matches the other constants in that file and I don't know ruby formatting rules. lmk if it should be all-caps and/or somewhere else.

That works for me. In ruby, anything that starts with a capital letter is a constant. E.g. class names are actually constants. I'm in the habit of using all caps just b/c it's been consistent across the programming languages i've used, but i don't really care at the moment. At some point, we should have a discussion if we want to enforce a consistent constant naming format.

**Comment by sadleb** [comment]
> > @sadleb I used `ContentLibraryCourseID` because it matches the other constants in that file and I don't know ruby formatting rules. lmk if it should be all-caps and/or somewhere else.
> 
> That works for me. In ruby, anything that starts with a capital letter is a constant. E.g. class names are actually constants. I'm in the habit of using all caps just b/c it's been consistent across the programming languages i've used, but i don't really care at the moment. At some point, we should have a discussion if we want to enforce a consistent constant naming format.

And as for location, there makes sense for now. We can refactor to a more appropriate place if we start needing to use that constant elsewhere in ruby code.

File: lib/canvas_api.rb [hunk 1]
```diff
@@ -157,4 +160,28 @@ def get_all_from_pagination(response)
     info
   end
 
+  # See: https://canvascoach.instructure.com/doc/api/file.file_uploads.html#method.file_uploads.post
+  # If successful, URL will be available as the hash value of the 'url' key.
+  # If unsuccessful, returns a Response object.
```
Comment (rshipp) [inline lib/canvas_api.rb:165]: I'm not happy with this interface, but I'm not sure what the "correct" way to do it is. seems like it should return the same kind of object whether or not it fails.

File: app/controllers/file_upload_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class FileUploadController < ApplicationController
+  # POST /course_contents.json
```
Comment (jessouyangbraven) [inline app/controllers/file_upload_controller.rb:2]: Is this supposed to be `#POST /file_upload.json`?

File: lib/canvas_api.rb [hunk 3]
```diff
@@ -18,6 +18,9 @@ class CanvasAPI
     </div>
   )
 
+  # Canvas course ID for the Braven Content Library.
+  ContentLibraryCourseID = 1
```
Comment (jessouyangbraven) [inline lib/canvas_api.rb:22]: Will we need different IDs for Booster? Should this eventually be something we configure in `.env`?

Comment (rshipp) [reply lib/canvas_api.rb:22]: afaik contentlibrary (which is part of Braven's custom modifications to our Canvas fork) is always course 1, and the editor will only ever need to publish to contentlibrary. @sadleb correct me if I'm wrong?

File: app/controllers/file_upload_controller.rb (hunk 2 above)
Comment (rshipp) [reply app/controllers/file_upload_controller.rb]: yep, thanks for the catch

**Pull request merged**